src/main/scala/firrtl/options/StageAnnotations.scala

```diff
@@ -40,3 +40,36 @@ case class TargetDirAnnotation(dir: String = ".") extends NoTargetAnnotation wit
     .unbounded() // See [Note 1]
     .text(s"Work directory for intermediate files/blackboxes, default is '.' (current directory)")
 }
+
+/** Additional arguments
+  *  - set with any trailing option on the command line
+  * @param value one [[scala.String]] argument
+  */
+case class ProgramArgsAnnotation(value: String) extends NoTargetAnnotation with StageOption {
+  def addOptions(p: OptionParser[AnnotationSeq]): Unit = p.arg[String]("<arg>...")
+    .unbounded()
+    .optional()
+    .action( (x, c) => c :+ ProgramArgsAnnotation(x) )
+    .text("optional unbounded args")
+}
+
+object ProgramArgsAnnotation {
+  private [options] def apply(): ProgramArgsAnnotation = ProgramArgsAnnotation("")
+}
```

Comment:
Better way to do this would be to make `HasScoptOptions` mixed into the companion object?

Reply:
@jackkoenig, you're 100% right about this approach. In the updated PR, `HasScoptOptions`, for `Annotation`s, is only mixed into the companion object.